Add the missing parity options for Windows

KathrynBaker opened the pull request:
There was a need for space parity for one device, it was not much effort to include mark as well.

At present this has been added only to the windows version as Linux does not use Space or Mark parity, and has some interesting work arounds to allow them to be used.

Testing during development was undertaken by using an existing asyn ioc and changing the parity type, then running it. It was verified that the value was being set by connecting to the moxa hosting the COM port being used by the IOC, and checking the async-Settings (Moxa page > Monitor > Async-Setting) were updated as appropriate. Please note that the test required a make of support/asyn, support/ioc and epics/ioc/master.

see ISISComputingGroup/IBEX#2461

KathrynBaker committed:
Add the nissing parity options for Windows

FreddieAkeroyd commented:
mark and space options should also be added to   menu(serialPRTY)  in  asynRecord.dbd
 
It looks like mark/space parity is not defined by POSIX, but many Linuxes support the CMSPAR parameter https://viereck.ch/linux-mark-space-parity/

KathrynBaker committed:
Add to dbd and linux options

KathrynBaker commented:
Added the options to the menu in the dbd, and added CMSPAR into the Linux side of things, in what I understood to be the appropriate fashion. Unable to test the Linux side of things, as I don't have access to such a system.

FreddieAkeroyd committed:
Check CMSPAR existance on Linux
